runtime-sdk/modules/contracts/src/results.rs

```diff
@@ -204,7 +205,7 @@ fn process_subcalls<Cfg: Config, C: TxContext>(
 
                 // Forward any emitted tags.
                 for tag in tags {
-                    ctx.emit_tag(tag);
+                    ctx.emit_tag(tag.key, tag.value.into());
```

Comment:
This doesn't seem like it will do the right thing (e.g. forward the value unchanged) because there is an interim conversion into a `Tag`. We can make `commit` return something else (e.g. a `Vec<(Vec<u8>, cbor::Value)>`) as well to handle this case?

Reply:
Yeah i think it makes sense, will also need to move the converting into tags into the dispatcher

Reply:
Given that this tuple now appears in multiple places we could also add a type like `EventTag` that would have the key and CBOR value pair.